src/ConstructiveSolidGeometry/PointsAndVectors/Vectors.jl

```diff
@@ -21,6 +21,35 @@ struct CartesianVector{T} <: AbstractCoordinateVector{T, Cartesian}
     z::T
 end
 
+#Type promotion happens here
+function CartesianVector(x::TX, y::TY, z::TZ) where {TX,TY,TZ}
+    eltypes = _csg_get_promoted_eltype.((TX,TY,TZ))
+    T = float(promote_type(eltypes...))
+    CartesianVector{T}(T(x),T(y),T(z))
+end
+
+function CartesianVector(x::TX, y::TY, z::TZ) where {TX<:Int,TY<:Int,TZ<:Int}
+    eltypes = _csg_get_promoted_eltype.((TX,TY,TZ))
+    T = float(promote_type(eltypes...))
+    CartesianVector{T}(T(x),T(y),T(z))
+end
```

Comment:
Why do you need two functions? Shouldn't the upper one do what the lower one does?

Reply:
Same as for the `CartesianPoint` I guess.